src/js/components/Modal.jsx

```diff
@@ -63,10 +65,18 @@ var Modal = React.createClass({
     },
 
     onSettingChange(settingKey, settingValue) {
-        if (settingKey === 'theme') {
-            UserPreferencesActions.set(UserPreferencesTypes.THEME, settingValue);
-        } else if (settingKey === 'showFiltersDescription') {
-            UserPreferencesActions.set(UserPreferencesTypes.SHOW_FILTERS_DESCRIPTION, settingValue);
+        switch (settingKey) {
```

Comment:
You should define the key inside switch, then call the `UserPreferencesActions.set()` method